Dockerfile

```diff
@@ -11,21 +11,28 @@ RUN ["bash", "-c", "yum install -y --setopt=tsflags=nodocs openssh-server libicu
      useradd -r -s /bin/bash -g git git && \
      mkdir --parent /home/git"]
 
-# https://gitlab.com/gitlab-org/gitlab-ce/issues/3027
-# https://docs.docker.com/engine/examples/running_ssh_service/
-RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
-
 # gitlab-shell setup
 COPY . /home/git/gitlab-shell
 WORKDIR /home/git/gitlab-shell
 
-COPY authorized_keys /home/git/.ssh/authorized_keys
 
-RUN ["bash", "-c", "mkdir ../gitlab-config && \
-                     cp config.yml.example ../gitlab-config/config.yml && \
-                     ln -s ../gitlab-config/config.yml && \
-                     ./bin/install && \
-                     chown -R git:git /home/git"]
+RUN mkdir /home/git/gitlab-config && \
+    ## Setup default config placeholder
+    cp config.yml.example ../gitlab-config/config.yml && \
+    ln -s /home/git/gitlab-config/config.yml && \
+    # PAM workarounds for docker and public key auth
+    sed -i \
+          -e "s|session\s*required\s*pam_loginuid.so|session optional pam_loginuid.so|g" \
+          -e "s|account\s*required\s*pam_nologin.so|#account optional pam_nologin.so|g" \
```

Comment:
So this stuff doesn't mean a lot to me. I think a link to some doc explanation would be good so we understand what is happening here